### PR TITLE
verifying statement before creating or modifying policy

### PR DESCRIPTION
**Pull request by Bento007**
* added typing.optional
* if no statement is give a default statement is used.
* removed extra variables from User
* adding test help functions for statements
* updating test cases to use valid statements
* adding tests case for invalid statements

### CODE REVIEW

File: tests/test_users.py [hunk 1]
```diff
@@ -88,14 +88,20 @@ def test_remove_groups(self):
             self.assertEqual(len(user.groups), 2)
 
     def test_set_policy(self):
-        name = "test_sete_policy@test.com"
+        name = "test_set_policy@test.com"
         user = User(self.directory, name)
         with self.subTest("The initial user policy is default_policy, when the user is first created"):
-            self.assertEqual(user.lookup_policies(), [self.default_policy])
+            self.assertEqual(user.statement, self.default_policy)
```
Comment (DailyDreaming) [inline tests/test_users.py]: Should `user.lookup_policies()` still be tested for consistency?

File: policies/default_user_policy.json [hunk 2]
```diff
@@ -2,12 +2,12 @@
   "Version": "2012-10-17",
   "Statement": [
     {
-      "Sid": "user_example",
-      "Effect": "Allow",
+      "Sid": "DefaultRole",
```
Comment (DailyDreaming) [inline policies/default_user_policy.json]: Nit: I think this is a copy-paste leftover from `policies/default_role.json` and should be `DefaultPolicy`.

File: tests/test_users.py (hunk 1 above)
Comment (Bento007) [reply tests/test_users.py]: This is tested else where.

File: fusillade/clouddirectory.py [hunk 3]
```diff
@@ -18,7 +18,7 @@
 
 project_arn = "arn:aws:clouddirectory:us-east-1:861229788715:"  # TODO move to config.py
 cd_client = boto3.client("clouddirectory")
-
+iam = boto3.client("iam")
```
Comment (xbrianh) [inline fusillade/clouddirectory.py]: If this code is executed from a Lambda, consider putting this behind an `lru_cache`

File: fusillade/clouddirectory.py [hunk 4]
```diff
@@ -678,6 +684,15 @@ def get_attributes(self, attributes: typing.List[str]):
             attrs[attr['Key']['Name']] = attr['Value'].popitem()[1]   # noqa
         return attrs
 
+    @staticmethod
+    def _verify_statement(statement):
```
Comment (xbrianh) [inline fusillade/clouddirectory.py]: Consider adding a docstring explaining the level of statement verification.

For instance, `iam.simulate_custom_policy` uses fake data for `ActionNames` and `ResourceArns`. Does this indicate that the statement is only syntactically verified?

Comment (Bento007) [reply fusillade/clouddirectory.py]: adding documentation

File: fusillade/clouddirectory.py (hunk 3 above)
Comment (Bento007) [reply fusillade/clouddirectory.py]: Adding todos

**Comment by Bento007** [comment]
@xbrianh Most of those questions will be answered when I finish swagger API.

File: policies/default_user_policy.json (hunk 2 above)
Comment (Bento007) [reply policies/default_user_policy.json]: fixed

File: tests/test_users.py (hunk 1 above)
Comment (Bento007) [reply tests/test_users.py]: added test anyways

**Pull request merged**